eva/catalog/models/base_model.py

```diff
@@ -113,13 +113,31 @@ def init_db():
         BaseModel.metadata.create_all()
 
 
-def clear_db_contents():
-    """Drop all of the record from tables."""
+def truncate_catalog_tables():
+    """Truncate all the catalog tables"""
     # https://stackoverflow.com/questions/4763472/sqlalchemy-clear-database-content-but-dont-drop-the-schema/5003705#5003705 #noqa
     engine = SQLConfig().engine
+    # reflect to refresh the metadata
+    BaseModel.metadata.reflect(bind=engine)
     if database_exists(engine.url):
         with contextlib.closing(engine.connect()) as con:
             trans = con.begin()
             for table in reversed(BaseModel.metadata.sorted_tables):
-                con.execute(table.delete())
+                if table.exists(con):
+                    con.execute(table.delete())
+            trans.commit()
+
+
+def drop_all_tables_except_catalog():
+    """drop all the tables except the catalog"""
+    engine = SQLConfig().engine
+    # reflect to refresh the metadata
+    BaseModel.metadata.reflect(bind=engine)
+    if database_exists(engine.url):
+        with contextlib.closing(engine.connect()) as con:
+            trans = con.begin()
+            for table in reversed(BaseModel.metadata.sorted_tables):
+                if table.name not in CATALOG_TABLES:
+                    if table.exists(con):
+                        table.drop(con)
```

Comment:
Wondering, we don't need `con.execute` here?

Reply:
It does not work for `drop` so had to do this way.